### PR TITLE
Add sandbox Dockerfile for running python code

### PR DESCRIPTION
**Pull request by jasonrhaas**
## What's new

This fixes issue #183.  I added a new Dockerfile that has a pre-built environment for running all of our Python code.  Now we can run the python code inside this container without installing Miniconda locally.  This should make it easier to get our code up and running quickly.

## Testing it out

In the `housing-insights` directory, 
- In your **secrets.json** file, change the `docker_database` string to work inside the docker container. 
```
    "docker_database": {
        "_comments": "You may need to change the IP address and/or port for your machine.",
        "connect_str": "postgresql://codefordc:codefordc@postgres:5432/housinginsights_docker"
    },
```

- `docker-compose up -d`.  This will bring up all the containers including the `sandbox` which is the miniconda3 python container.
- `docker-compose exec sandbox bash` to get into the docker container
- `cd /code/python/scripts` to get into the script directory to load the data
- `python load_data.py docker rebuild sample` to load in the sample data

**NOTE**:  The data loading does not work at the moment due to some import errors.  This error is the same in the local environment and docker environment so it should not be an issue with the container.

### CODE REVIEW

File: docs/resources/onboarding/docker.md [hunk 1]
```diff
@@ -54,6 +64,14 @@ This will print the IP address of your local Docker instance. Open up `/python/h
...
+
+```
+    "docker_database": {
+        "connect_str": "postgresql://codefordc:codefordc@localhost:5432/housinginsights_docker"
+    },
+```
+
 **Troubleshooting note**: If you already have Postgres installed on your machine, it may be using the 5432 port. If you get an error message saying something related to a port in use, you'll need to change the port both here AND in the docker-compose.yml file. 
 
 ### 1b) Day-to-day use
```
Comment (NealHumphrey) [inline docs/resources/onboarding/docker.md]: I think the line 90 below (new numbering) should change to `docker-compose up -d` to allow access to the docker command line, right?

File: docs/resources/onboarding/docker.md [hunk 2]
```diff
@@ -87,6 +105,30 @@ If you've got Docker running, you should be able to see your database in the web
 
 ### 2a) One time setup
 
+Here again there is the option to run python inside the `docker-compose` environment or using your local python setup.  The easiest method is to run using the first option below, but the others will work as well.
+
+**Full Docker Compose (docker Python)**
+
+1)  `docker-compose ps` to see the containers that are running.  One of them should be **housinginsights_sandbox_1**.  This is the container that is running the miniconda3 python environment.  There is no need to activate or deactivate an virtualenv since it is already sourced.
```
Comment (NealHumphrey) [inline docs/resources/onboarding/docker.md]: Can you add a note clarifying that this is the same command line (or Docker Quickstart Terminal when using Toolbox) as the one that you ran docker-compose up on? I expect there will be confusion since the local option (which is what people are used to using) used two separate terminal windows.

**Comment by NealHumphrey** [comment]
@jasonrhaas Thanks for this! I'm going to have to wait until tomorrow to test this locally but should get this in tomorrow. Documentation looks overall great, there's just the one note I had about clarifying where to type the commands (single vs. two command prompt windows). Can you verify my interpretation was correct and edit if so?

File: docs/resources/onboarding/docker.md (hunk 1 above)
Comment (xee5ch) [inline docs/resources/onboarding/docker.md]: If you run `docker-compose up -d` I was under the impression it runs it all in the background; the `-d` is for detached after all.

**Comment by xee5ch** [comment]
Again, I am not sure if I am doing this wrong, but this is exactly the same issue I ran into previously when trying to build a docker-compose worfklow using `build` with `Dockerfile` directives with manually copied files (in your case environment.yml and requirements.txt are in properly in `/usr/src/app` as desired) and `volume` directives (in your case `/code` is empty: you either get the one or the other, but not both.

```
[me@craptop housing-insights]$ docker-compose down
[me@craptop housing-insights]$ docker rmi SANDBOXHASH # just to see I am not crazy
[me@craptop housing-insights]$ docker-compose build
[me@craptop housing-insights]$ docker-compose run sandbox bash
Creating network "housinginsights_default" with the default driver
root@ce721b10be2e:/usr/src/app# 
root@ce721b10be2e:/usr/src/app# pwd
/usr/src/app
root@ce721b10be2e:/usr/src/app# ls
environment.yml  requirements.txt
root@ce721b10be2e:/usr/src/app# ls -la /code/
total 4
drwxr-xr-x  3 root root   60 May  9 01:50 .
drwxr-xr-x 47 root root 4096 May  9 02:14 ..
drwxr-xr-x  2 root root   40 May  9 01:50 docs
root@ce721b10be2e:/usr/src/app# ls -la /code/docs/
total 0
drwxr-xr-x 2 root root 40 May  9 01:50 .
drwxr-xr-x 3 root root 60 May  9 01:50 ..
root@ce721b10be2e:/usr/src/app# 
```

Correct me if I am wrong, but we want the python code added to the container, no? I get the feeling I must be doing something wrong, but I found lots of warnings about workdir and volume overlay not going as planned, it's just not clear if it is with docker-machine or not. I used machine on purpose to make sure I get the same environment as the Mac and Windows people as well, so as I discussed with Neal on Sunday, I feel I am beating a dead horse or I am misreading docker-compose/machine docs wrong.

File: docs/resources/onboarding/docker.md (hunk 1 above)
Comment (jasonrhaas) [inline docs/resources/onboarding/docker.md]: Oh yea, I always run `docker-compose up -d`, and do `docker-compose logs <name>` to access the logs if I need to.  If things are running properly, I usually don't care about seeing the terminal output.

I can change the command all to `docker-compose up -d` if you are good with that.

File: docs/resources/onboarding/docker.md (hunk 2 above)
Comment (jasonrhaas) [inline docs/resources/onboarding/docker.md]: I think we should just change everything to `docker-compose up -d` and then we don't have to worry about the confusion.

**Comment by jasonrhaas** [comment]
@xee5ch Are you running all the latest versions of Docker and docker-compose?  Are you running on Windows, Mac, or Linux?  I am sure that the above implementation is working properly on Mac and Linux, but I have not tried it on Windows.  Maybe the volume mounts work a little differently with Windows, or maybe its a system config thing?

As far as copying the python code over to the image, we shouldn't do this unless we are running production code inside the container.  This container is only used for development purposes -- copying code over makes it difficult to do development on, because anytime you make a change to the repo code you need to rebuild the container, which takes time and is easy to forget.  And you may find your self in a situation where you think you are running the latest code inside the container, but really its old code, and stuff is broken and then you waste time trying to figure out why.

**Comment by xee5ch** [comment]
You're all good. It turns out the Linux version of docker-machine, of using the underlying VBox defaults even if guest support is configured (I did so) fails at filesystem sharing. That was the root cause all along.

I stayed up late and went back to normal Docker on Linux and solved networking issues I had. I was trying to stay on the same page as everyone else. Sorry for wasting time.

TL;DR docker-machine on Linux cannot share /home like it does /Users on on OSX or Windows in many cases so its looking for the files in the docker-machine VM. This is a known limitation and those like me should know better.

I tested your stuff and it works well.

**Comment by jasonrhaas** [comment]
Ahhh yea the old guest support thing with Virtual Box.  As soon as native support for Docker was available for Mac, I switched to that and never dealt with VM's again.  Same thing with Linux, just run Docker natively, and it makes things a lot easier.  While Docker doesn't solve everything, it has _almost_ killed the need to test in VM's, which is 💯 in my book.

File: docs/resources/onboarding/docker.md (hunk 1 above)
Comment (jasonrhaas) [inline docs/resources/onboarding/docker.md]: @NealHumphrey are you good with the above comment?

**Comment by NealHumphrey** [comment]
Thanks @jasonrhaas ! Tested and works great. I'm making a couple more documentation edits and will tell everyone about the new option in the update email today.
